File: FuelSample/Objective-C/FuelSample/FuelResultVenderViewController.m
//
//  FuelResultVenderViewController.m
//  FuelSample
//
//  Copyright © 2559 Globtech. All rights reserved.
//

#import "FuelResultVenderViewController.h"

@implementation FuelResultVenderViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    NTFuelParameter *param = nil;
    
    if (_adminLevel1 && _adminLevel2) {
        param = [[NTFuelParameter alloc] initWithAdminLevel1:_adminLevel1
                                                 adminLevel2:_adminLevel2];
    }
    else {
        param = [[NTFuelParameter alloc] initWithLat:_lat lon:_lon];
    }
    
    NSError *error = nil;
    
    NTFuelResultSet *resultSet = [NTFuelService execute:param error:&error];
    
    if (!error) {
        _results = resultSet.results;
    }
    
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([segue.identifier isEqualToString:@"priceSegue"]) {
        FuelResultPriceViewController *fuelResultPriceViewController = segue.destinationViewController;
        NTFuelResult *result = sender;
        fuelResultPriceViewController.title = result.brandName_L;
        fuelResultPriceViewController.result = result;
    }
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    NTFuelResult *result = _results[indexPath.row];
    [self performSegueWithIdentifier:@"priceSegue" sender:result];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell"];
    NTFuelResult *result = _results[indexPath.row];
    
    cell.textLabel.text = result.brandName_L;
    
    return cell;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return _results != nil ? _results.count : 0;
}

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

@end

File: FuelSample/Swift/FuelSample/FuelResultVenderViewController.swift
//
//  FuelResultViewController.swift
//  FuelSample
//
//  Created by Itthisak Phueaksri on 5/15/2559 BE.
//  Copyright © 2559 gissoft. All rights reserved.
//

import UIKit
import NOSTRASDK

class FuelResultVendorViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {

    var adminLevel1: String?;
    var adminLevel2: String?;
    
    var lat: Double?;
    var lon: Double?;
    
    var results: [NTFuelResult]?;
    
    override func viewDidLoad() {
        super.viewDidLoad();
        
        var param: NTFuelParameter! = nil;
        
        if adminLevel1 != nil && adminLevel2 != nil {
            param = NTFuelParameter(adminLevel1: adminLevel1!, adminLevel2: adminLevel2!);
        }
        else {
            param = NTFuelParameter(lat: lat!, lon: lon!);
        }
        
        do {
            let resultSet = try NTFuelService.execute(param);
            results = resultSet.results;
        }
        catch let error as NSError {
            print("error: \(error.description)");
        }
        
    }
    
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        if segue.identifier == "priceSegue" {
            let priceViewController = segue.destinationViewController as? FuelResultPriceViewController;
            let result = sender as! NTFuelResult;
            priceViewController?.title = result.brandName_L;
            priceViewController?.result = result
        }
    }
    
    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        let result = results![indexPath.row];
        
        self.performSegueWithIdentifier("priceSegue", sender: result);
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier("cell");
        let result = results![indexPath.row];
        
        cell?.textLabel?.text = result.brandName_L;
        
        
        return cell!;
    }
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return results != nil ? (results?.count)! : 0;
    }
    
    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 1;
    }
    
}
